### PR TITLE
[alma8] Add rpm command to dump information about the packages in the container

### PR DESCRIPTION
**Pull request by aandvalenzuela**
@smuzaffar, I would like to reuse the `rpm-pkg-info.sh` script in all config files, can I put it one level up and get the repo name using `basename`?

### CODE REVIEW

File: alma8/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM @BASE_IMAGE_NAME@
-RUN dnf install -y @DEFAULT_PACKAGES@ automake bash bzip2 bzip2-libs bzip2-devel \
+RUN dnf install -y bash && bash @SCRIPT@ BASE &&\
```
Comment (smuzaffar) [inline alma8/Dockerfile:2]: @aandvalenzuela , why using `@SCRIPT@`? just add
```
ADD rpm-pkg-info.sh /tmp/rpm-pkg-info.sh
```
after `FROM ` command. and then use `/tmp/rpm-pkg-info.sh` to run it.

You can move the  `rpm-pkg-info.sh` one level up but still need to create a symlink in the repo directory e.g.
```
cms-docker/etc/rpm-pkg-info.sh
cms-docker/alma8/rpm-pkg-info.sh -> ../etc/rpm-pkg-info.sh  (symlink)
```

File: alma8/Dockerfile [hunk 2]
```diff
@@ -36,9 +36,9 @@ RUN dnf install -y bash && bash /tmp/rpm-pkg-info.sh BASE &&\
     echo el8 > /etc/cmsos &&\
     dnf clean all
 
-ADD krb5.conf /etc/krb5.conf
-ADD ca-bundle.crt /etc/pki/tls/certs/ca-bundle.crt
-ADD fix_ssh_config.sh /tmp/fix_ssh_config.sh
+ADD @DOCKER_NAME@/krb5.conf /etc/krb5.conf
+ADD @DOCKER_NAME@/ca-bundle.crt /etc/pki/tls/certs/ca-bundle.crt
+ADD @DOCKER_NAME@/fix_ssh_config.sh /tmp/fix_ssh_config.sh
```
Comment (smuzaffar) [inline alma8/Dockerfile:41]: these are common scripts/configurations and can also go in some common directory e.g. `etc` or `share`. So I would suggest to copy these to `share` and then use `share/filename` in the Docker file

File: alma8/config.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ variables:
   CMS_COMPATIBLE_OS: el8
   DEFAULT_PACKAGES: coreutils-single
   EPEL_PACKAGES:
+  DOCKER_NAME: alma8
```
Comment (smuzaffar) [inline alma8/config.yaml]: by the way, no need to add `DOCKER_NAME` , this is already set by default for each container build job

**Comment by aandvalenzuela** [comment]
Thanks @smuzaffar! It builds locally, should I modify the others in a follow-up PRs? Otherwise we will trigger rebuilds for all of the images, I think

**Comment by smuzaffar** [comment]
@aandvalenzuela , with this change , you have to make sure that code section https://github.com/cms-sw/cms-docker/blob/master/bin/get_image_config.py#L140-L145 also work. This code reads the file pointed by `ADD or COPY`command and include it in checksum. Currently it assumes that file exists under repo directory. So you need to add the concept of `build_context` in the config.yaml. Default should be `.` and amla8/config.yaml can set it to `..` and then above mention code can make use of it to read the `ADD/COPY` files contents

**Pull request closed without merging**